### PR TITLE
feat(clients): add waiters to aws clients

### PR DESCRIPTION
**Pull request by alexforsyth**
*Issue #, if available:*
depends on https://github.com/awslabs/smithy-typescript/pull/243
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: clients/client-accessanalyzer/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "remove-definitions": "rimraf ./types",
     "remove-dist": "rimraf ./dist",
     "remove-documentation": "rimraf ./docs",
-    "test": "exit 0",
+    "test": "yarn build && jest --coverage --passWithNoTests",
```
Comment (AllanZhengYP) [inline clients/client-accessanalyzer/package.json:13]: Is this generated from `yarn generate-clients`? It should not update the package.json. Can you revert the changes to package.json files?

Comment (alexforsyth) [reply clients/client-accessanalyzer/package.json]: yeah this is old, dont know exactly why this is happening

**Comment by alexforsyth** [comment]
This will fail until https://github.com/aws/aws-sdk-js-v3/pull/1759 is merged

**Review by AllanZhengYP — APPROVED** [review]
